mcs/class/System.Private.CoreLib/System.Runtime.InteropServices/Marshal.cs

```diff
@@ -49,7 +49,7 @@ internal static IntPtr AllocBSTR (int length)
 
 		internal static bool IsPinnable (object obj)
 		{
-			throw new NotImplementedException ();
+			return (o == null) || RuntimeTypeHandle.HasReferences (obj.GetType () as RuntimeType));
 		}
```

Comment:
its called 'obj' not 'o'.

Reply:
is it ever called with null?

Reply:
@marek-safar why not? `gcHandle.Target = null;` or `GCHandle.Alloc(null, ...);`